Fix: Changes tf message type to tf2.

matthias-mayr opened the pull request:
We noticed that the MIR is publishing messages in tf2. This is one way that works from us, but one might want to use one of those rewrite functions.

I am not too familiar with this software stack, so I leave it to you.

mintar commented:
This is only a cosmetic change. `tf.msg.tfMessage` and `tf2_msgs.msg.TFMessage` have the same md5sum, so it doesn't matter which one you publish; if one node publishes tf and another subscibes to tf2, it just works.

I have no objections against switching to tf2, but in that case also the `package.xml` and `tf_remove_child_frames.py` should be updated. If you want to add it to this PR, I'll merge it.

matthias-mayr committed:
Fix: Adds subscription to "tf_static".

Some transformations are published on this topic and are needed to
obtain a full tf tree. E.g. "base_footprint" to "base_link"

matthias-mayr commented:
Thanks for the comments. They were exactly the type of comment I wanted to have on this PR.

I removed the change of the tf message include and added only the additional subscription that is necessary to get a full tf tree.

mintar commented:
Ah, I hadn't seen that part of the PR. Yes, adding `tf_static` is of course necessary. Thanks for the fix!
